### ISSUE TITLE
Using clink-gitprompt in non get directories is slow

### ISSUE THREAD

**Issue by idvorkin** [djs/clink-gitprompt#3]
Because the code calls into git regardless of the directory you're in, there's noticeable delay on every prompt.  Could the code run a heuristic to test if we're likley in a git dir by checking for a ".git" directory first? Something like: 

```
 local possibleGitDir = ".git"
    for i=0, 10 do
       if directory_exists(possibleGitDir) then
            return gitDir
       end 
          possibleGitDir = "../"..possibleGitDir
      end
     return false
 end
```

**Comment by djs** [comment]
Sounds like a reasonable idea. Feel free to try it out and send a pull request. I don't really use this anymore because of the speed problem and I prefer to use git bash. My long-term idea was to call into libgit2 since I actually care mostly about the speed of the prompt while I am inside a git repository.

**Comment by idvorkin** [comment]
I agree libgit2 is a better approach, but beyond the time I can invest now. Lets see if we get others interested. If someone will drive I'm happy to do some testing/bugfixes.

**Comment by piotr-cz** [comment]
There is a repo called [clink-completions](https://github.com/vladimir-kotikov) with git extension which consists an [get_git_dir](https://github.com/vladimir-kotikov/clink-completions/blob/master/git.lua#L12) which looks up `.git` directory up to the drive's root.

**Comment by piotr-cz** [comment]
Aforementioned function ported to clink-gitprompt makes huge difference in performance.
I don't think there is a point of using libgit2 for this particular operation.

**Comment by djs** [comment]
Can you submit a pull request so I can try it out?

**Comment by piotr-cz** [comment]
Yes I can, but besides function `get_git_dir` from `git.lua` I've also used [`modules/path.lua`](https://github.com/vladimir-kotikov/clink-completions/blob/master/modules/path.lua) and [`.init.lua`](https://github.com/vladimir-kotikov/clink-completions/blob/master/.init.lua) to load it.
